### PR TITLE
⭐ hetzner: parse TLS certificate PEM for hygiene audits

### PR DESCRIPTION
**Pull request by tas50**
Parses the TLS certificate PEM the `hetzner.certificate` resource already stores and surfaces eight fields for certificate-hygiene audits. All are extracted via `crypto/x509` from the in-hand PEM — **no new API call, no new IAM permission**. Follow-up to #8979 / #8982.

## Fields

| Field | Type | Enables |
|---|---|---|
| `keyAlgorithm` / `keyBits` | `string` / `int` | RSA `< 2048`, legacy key types |
| `signatureAlgorithm` | `string` | flag `SHA1-RSA` / `MD5-RSA` |
| `issuer` / `subject` / `serialNumber` | `string` | provenance, pinning |
| `isCa` | `bool` | leaf vs CA |
| `selfSigned` | `bool` | self-signed on internet-facing services |

```mql
hetzner.certificates.where( signatureAlgorithm == /SHA1|MD5/ )
hetzner.certificates.where( keyAlgorithm == "RSA" && keyBits < 2048 )
hetzner.certificates.where( selfSigned )
```

Zero values (`""`, `0`, `false`) are returned when the PEM body is absent or unparseable, so audits can distinguish "unknown" from a real value.

## Testing

- New table-driven tests for the parser: RSA-2048 / ECDSA-P256 / Ed25519 self-signed, a CA-signed leaf (asserts `selfSigned == false`), and empty / non-PEM / valid-PEM-wrapping-junk inputs. Certs are generated in-test via `crypto/x509.CreateCertificate` (no hardcoded blobs).
- Also backfilled a test for the existing `targetServerIDs` load balancer helper (dedup + nil-skip).
- `go build ./...`, `go vet`, `gofmt`, and the full provider suite pass.

Not smoke-tested against a live Hetzner project (no API token in the dev environment); the parser is unit-tested against generated certs.

### CODE REVIEW

File: providers/hetzner/resources/hetzner_certificate.go [hunk 1]
```diff
@@ -16,6 +22,55 @@ type mqlHetznerCertificateInternal struct {
...
+		return out
+	}
+	out.keyAlgorithm = c.PublicKeyAlgorithm.String()
+	out.signatureAlgorithm = c.SignatureAlgorithm.String()
+	out.issuer = c.Issuer.String()
+	out.subject = c.Subject.String()
+	out.isCa = c.IsCA
+	// A self-issued (self-signed) certificate has an identical issuer and
+	// subject; comparing the raw DER is the canonical check.
+	out.selfSigned = bytes.Equal(c.RawIssuer, c.RawSubject)
```
[the remark this answers is not in the document]
Comment (tas50) [reply providers/hetzner/resources/hetzner_certificate.go]: Fixed in the latest commit. `selfSigned` now requires both self-issued names *and* that the signature verifies against the certificate's own public key (`CheckSignature`), so a cross-signed cert with matching issuer/subject names is no longer reported as self-signed. I kept `CheckSignature` over `CheckSignatureFrom` because the latter also requires the parent to be a CA and would false-negative on a self-signed non-CA leaf. Added a cross-signed test case covering it.
